File: app/src/main/java/com/mxt/anitrend/api/structure/Episode.java
package com.mxt.anitrend.api.structure;

import android.os.Parcel;
import android.os.Parcelable;

import org.simpleframework.xml.Element;
import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Root;

import java.util.List;

/**
 * Created by Maxwell on 2/9/2017.
 */
@Root(name = "item", strict = false)
public class Episode implements Parcelable {

    @Element(name = "title")
    private String title;
    @Element(name = "link")
    private String link;
    @Element(name = "description")
    private String description;
    @Element(name = "publisher", required = false)
    private String publisher;
    @Element(name = "content")
    private MediaContent content;
    @ElementList(name = "thumbnail", inline = true, required = false)
    private List<Thumbnail> thumbnail;

    public Episode(@Element(name = "title")String title, @Element(name = "link")String link, @Element(name = "description")String description, @Element(name = "publisher", required = false)String publisher, @Element(name = "content")MediaContent content, @ElementList(name = "thumbnail", inline = true, required = false)List<Thumbnail> thumbnail) {
        this.title = title;
        this.link = link;
        this.description = description;
        this.publisher = publisher;
        this.content = content;
        this.thumbnail = thumbnail;
    }

    protected Episode(Parcel in) {
        title = in.readString();
        link = in.readString();
        description = in.readString();
        publisher = in.readString();
        content = in.readParcelable(MediaContent.class.getClassLoader());
        thumbnail = in.createTypedArrayList(Thumbnail.CREATOR);
    }

    public static final Creator<Episode> CREATOR = new Creator<Episode>() {
        @Override
        public Episode createFromParcel(Parcel in) {
            return new Episode(in);
        }

        @Override
        public Episode[] newArray(int size) {
            return new Episode[size];
        }
    };

    public String getTitle() {
        return title;
    }

    public String getLink() {
        return link;
    }

    public String getDescription() {
        description = description.replaceAll("(<img[^>]*>)","").replaceFirst("(<br[^>]*>)","");
        if(description.isEmpty())
            return title+" has no summary information at the moment.";
        return description;
    }

    public String getPublisher() {
        return publisher;
    }

    public MediaContent getContent() {
        return content;
    }

    public List<Thumbnail> getThumbnail() {
        return thumbnail;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(title);
        parcel.writeString(link);
        parcel.writeString(description);
        parcel.writeString(publisher);
        parcel.writeParcelable(content, i);
        parcel.writeTypedList(thumbnail);
    }
}

File: app/src/main/java/com/mxt/anitrend/model/entity/crunchy/Episode.kt
package com.mxt.anitrend.model.entity.crunchy

import android.os.Parcelable
import kotlinx.parcelize.Parcelize
import org.simpleframework.xml.Element
import org.simpleframework.xml.ElementList
import org.simpleframework.xml.Root

/**
 * Created by max on 2/9/2017.
 */
@Root(name = "item", strict = false)
@Parcelize
class Episode(
    @field:Element(name = "title")
    var title: String = "",
    @field:Element(name = "link")
    var link: String? = null,
    @field:Element(name = "description", required = false)
    @get:JvmName("getRawDescription")
    @set:JvmName("setRawDescription")
    var description: String = "",
    @field:Element(name = "publisher", required = false)
    var publisher: String? = null,
    @field:Element(name = "content", required = false)
    var content: MediaContent? = null,
    @field:ElementList(name = "thumbnail", inline = true, required = false)
    var thumbnail: List<Thumbnail> = mutableListOf()
) : Parcelable {

    fun getDescription(): String {
        val safeDescription = description
            .replace("(<img[^>]*>)".toRegex(), "")
            .replaceFirst("(<br[^>]*>)".toRegex(), "")
        return if (safeDescription.isEmpty()) {
            "$title has no summary information at the moment."
        } else {
            safeDescription
        }
    }
}
